### PR TITLE
Adds a path for Pure end to end tests

### PR DESCRIPTION
**Pull request by jayunit100**
This is a first pass at implementing a code path that
- runs upstream E2Es in such a way that all the flags and so on are properly consumed,
- no OS requirements are coupled into the test run.
- invokes the upstream test/e2e/e2e.go in the same way as upstream.

### CODE REVIEW

**Comment by jayunit100** [comment]
cc @timothysc @smarterclayton ... 
This is a first pass at implementing a code path that 
- runs upstream E2Es in such a way that all the flags and so on are properly consumed, 
- no OS requirements are coupled into the test run. 
- invokes the upstream test/e2e/e2e.go in the same way as upstream.

**Comment by jayunit100** [comment]
I sort of am considering forcing the check for the env var so its unambiguous...?

**Comment by jayunit100** [comment]
cc @timothysc @smarterclayton

File: test/extended/extended_test.go [hunk 1]
```diff
@@ -11,15 +11,28 @@ import (
...
 
 	exutil "github.com/openshift/origin/test/extended/util"
 )
 
 // init initialize the extended testing suite.
 func init() {
-	exutil.InitTest()
+	// Kubernetes: Pure end to end tests ~ These flags pass through.
+	if os.Getenv("RUN_KUBE_E2E_SUBSET") == "true" {
+		e2e.RegisterFlags()
```
Comment (liggitt) [inline test/extended/extended_test.go]: I don't think we want a command exposing completely different flags toggled by envvar... @smarterclayton do you envision the extended test being a binary that honors *exactly* the same flags as upstream e2e? If so, do we even need the envvar guard, and if not, would we be better off with a separate binary?

Comment (jayunit100) [inline test/extended/extended_test.go]: - we all agreed in a separate issue that we want an e2e binary of the kube e2es to ship alongside.
- the only issue as jordan suggests: Do we want a separate binary?  I think clayton said 'no' but we can wait for his response...

**Comment by jayunit100** [comment]
agree env vars are wonky.  other ideas? maybe a required flag `--testtype=upstream|extended` ?

File: test/extended/extended_test.go (hunk 1 above)
Comment (liggitt) [inline test/extended/extended_test.go]: he did say no, but I don't think he envisioned a single binary accomplished this way :)

Comment (jayunit100) [inline test/extended/extended_test.go]: /me eagerly awaits his thoughts :)

Comment (smarterclayton) [inline test/extended/extended_test.go]: I didn't envision a single binary accomplished this way.

If we need flags, we should add flags to extended.  We don't have to preserve the old env vars or flags as long as the new flags cover all the use cases.

File: test/extended/extended_test.go [hunk 2]
```diff
@@ -11,15 +11,28 @@ import (
...
+	"testing"
 
 	exutil "github.com/openshift/origin/test/extended/util"
 )
 
 // init initialize the extended testing suite.
 func init() {
-	exutil.InitTest()
+	// Kubernetes: Pure end to end tests ~ These flags pass through.
+	if os.Getenv("RUN_KUBE_E2E_SUBSET") == "true" {
```
Comment (timothysc) [inline test/extended/extended_test.go]: um why not just command line toggle.

File: test/extended/extended_test.go (hunk 1 above)
Comment (jayunit100) [inline test/extended/extended_test.go]: Is the only issue the env var ? I agree flag for this is better.

...Otherwise okay ?

File: test/extended/extended_test.go (hunk 2 above)
Comment (jayunit100) [inline test/extended/extended_test.go]: SGTM

**Comment by jayunit100** [comment]
Per IRC: Why not just use test filters? 
- e2e.go is the idiomatic way to call upstream tests. if we avoid that path, we have to chase down the logic,  its a very hot spot in the code.
Proposed solution: 
- commandline flag or Env var to turn on upstream only tests, rather than lumping them in with all the logic for initializing openshift tests.

**Comment by jayunit100** [comment]
simplifying, i think i can simplify this patch to do nothing other than upstream flag parsing

from there we can debate the finer points in  a separate pr

**Comment by jayunit100** [comment]
Ok, I think this is something we can all agree on for now... Parsing the flags~ NO CHANGE to code path, just flags.

PTAL @liggitt @smarterclayton @timothysc .

File: test/extended/extended_test.go [hunk 3]
```diff
@@ -11,15 +11,19 @@ import (
...
 
 // init initialize the extended testing suite.
 func init() {
+	// Kubernetes: Pure end to end tests ~ These flags pass through.
+	e2e.RegisterFlags()
 	exutil.InitTest()
 }
 
 func TestExtended(t *testing.T) {
+	// TODO: consider adding kube specific execution path that uses e2e.go library.
```
Comment (jayunit100) [inline test/extended/extended_test.go]: only question : how we're going to run these tests now...

**Comment by jayunit100** [comment]
This PR aims to solve #7454 .
Marked as do not merge until we decide on how to impl the call to the e2e tests.

**Comment by jayunit100** [comment]
As another impl, simply gutting the existing E2E Runner seems to work at first glance.
Testing now...

**Comment by jayunit100** [comment]
TL; DR 
I've gone with the (1st) solution: Update extended test to use upstream, rather than having 2 code paths.So far appears to be working for me...

**Comment by jayunit100** [comment]
WIP removed, as i think this should work in CI properly now.  So long as it doesnt adversly effect extended/networking.  I'm able to run openshift tests easily w/ `test/extended/core.sh ginkgo.focus=openshift`

File: test/extended/extended_test.go [hunk 4]
```diff
@@ -11,15 +11,19 @@ import (
...
 func init() {
+	// Kubernetes: Pure end to end tests ~ These flags pass through.
+	e2e.RegisterFlags()
 	exutil.InitTest()
 }
 
 func TestExtended(t *testing.T) {
-	exutil.ExecuteTest(t, "Extended Core")
+	// TODO Send a second arg once a name is supported upstream.
+	e2e.RunE2ETests(t)
```
Comment (timothysc) [inline test/extended/extended_test.go]: This looks much cleaner, thx.

**Comment by jayunit100** [comment]
hi @smarterclayton , any thoughts? i think this is the simplest solution to start converging the runner w/ upstream

**Comment by smarterclayton** [comment]
@marun can you look at the networking changes here?

**Comment by marun** [comment]
@smarterclayton Now that #8011 has merged and dropped test/extended/networking/extended_test.go, this change won't have any networking-specific impact.

**Comment by danmcp** [comment]
@jayunit100 Needs rebase
@smarterclayton Other thoughts?

**Comment by smarterclayton** [comment]
Assuming all the suites pass I don't see any reason not to merge.  Jay if you can rebase and kick the tests we'll merge.

**Comment by timothysc** [comment]
Closing this as it's been fixed a while ago.
